src/plugins/intel_gpu/src/graph/graph_optimizer/compile_graph.cpp

```diff
@@ -90,7 +90,8 @@ void compile_graph::run(program& p) {
 
         // TODO: need to come up with better handling of unsupported shape agnostic cases
         // e.g. process exceptions from choose_impl() and ignore those for dynamic parameters
-        if (node->is_type<fully_connected>() && node->is_dynamic() && node->get_output_pshape().size() > 3)
+        if (node->is_type<fully_connected>() && node->is_dynamic()
+            && (node->get_output_pshape().size() > 3 || node->get_preferred_impl_type() == impl_types::onednn))
```

Comment:
This is changing the original behavior not to use shape agnostic kernel always when it is onednn impl. Is that you intention?

Reply:
I modified the condition to merge two different conditions to one.

if (node->is_type<fully_connected>() && node->is_dynamic() && node->get_output_pshape().size() > 3) can_select_impl = false; // this is the original condition.
if (node->is_type<fully_connected>() && node->is_dynamic() && node->get_preferred_impl_type() == impl_types::onednn) can_select_impl = false; // I added it to disable choose impl when the FC is onednn impl type and it has dynamic shape.

Reply:
Since no onednn primitives support shape agnostics now, frankly we don't need to check FC condition.

How about add additional if states for setting can_select_iml to false when all onednn primitive with dynamic?

Reply:
Fixed it.